### ISSUE TITLE
ArgumentOutOfRangeException being thrown while editing, causing hard crashes

### ISSUE THREAD

**Issue by cartercanedy** [Exafunction/CodeiumVisualStudio#59]
### Stack trace:

> Application: devenv.exe
> Framework Version: v4.0.30319
> Description: The process was terminated due to an unhandled exception.
> Exception Info: System.ArgumentOutOfRangeException
>    at System.String.Substring(Int32, Int32)
>    at CodeiumVS.SuggestionTagger.AddInsertionTextBlock(Int32, Int32, System.String)
>    at CodeiumVS.SuggestionTagger.UpdateAdornment(Microsoft.VisualStudio.Text.Editor.IWpfTextView, System.String, Int32)
>    at CodeiumVS.SuggestionTagger.Update()
>    at CodeiumVS.CodeiumCompletionHandler+<GetCompletion>d__14.MoveNext()
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
>    at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
>    at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
>    at System.Threading.ThreadPoolWorkQueue.Dispatch()

### Context:
I think most of the crashes have occurred after the length of the buffer decreases.

**Comment by JustaSimpleUser** [comment]
I don't know if I have the same problem, but I'm seeing crashes with the latest releases of VS Community (v17.9.2) and Codeium (1.6.39).  I'm editing C++ code and IntelliCode has been disabled in Tools > Options > IntelliCode (all fields deselected).

Which version of VS and Codeium are you using?  From where did you get the stack trace?

**Comment by cartercanedy** [comment]
Stack trace came from windows event log

**Comment by kyhou** [comment]
yeah, I'm having the same problem, and some friends that I introduced to the extension too.

**Comment by M0n7y5** [comment]
Same for me
```csharp
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ArgumentOutOfRangeException
   at System.String.Substring(Int32, Int32)
   at CodeiumVS.CodeiumCompletionHandler.ParseCompletion(System.Collections.Generic.IList`1<CodeiumVS.Packets.CompletionItem>, System.String, System.String, System.String, Int32)
   at CodeiumVS.CodeiumCompletionHandler+<GetCompletion>d__14.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

**Comment by ianabrown** [comment]
I am getting the exact same issue as the OP, with the same stacktrace and hard crash of VS 2022 17.9.2 and Codeium extension 1.8.2. 
For me, it happened multiple times when I was doing some manual refactoring by cutting/pasting code from one class to another where it would lock up for a few seconds and then crash VS after pasting a code block. The problem didn't seem to occur during normal code creation.
`
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IndexOutOfRangeException
   at CodeiumVS.SuggestionTagger.UpdateAdornment(Microsoft.VisualStudio.Text.Editor.IWpfTextView, System.String, Int32)
   at CodeiumVS.SuggestionTagger.Update()
   at CodeiumVS.CodeiumCompletionHandler+<GetCompletion>d__14.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
`

**Comment by fortenforge** [comment]
I just published 1.8.14 which should fix all crashes that I am aware of. Please re-open this issue if you experience a crash after updating to this version.
